api/catalog/api/serializers/base.py

```diff
@@ -1,24 +1,25 @@
-import re
-
-from django.conf import settings
 from rest_framework import serializers
 
 
-class SchemableHyperlinkedIdentityField(serializers.HyperlinkedIdentityField):
+class BaseModelSerializer(serializers.ModelSerializer):
     """
-    This field returns the link but allows the option to replace the URL scheme.
+    Extends model serializer to use docstring of properties as help text.
     """
 
-    def __init__(self, scheme=settings.API_LINK_SCHEME, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        self.scheme = scheme
-
-    def get_url(self, *args, **kwargs):
-        url = super().get_url(*args, **kwargs)
-
-        # Only rewrite URLs if a fixed scheme is provided
-        if self.scheme is not None:
-            url = re.sub(r"^\w+://", f"{self.scheme}://", url, 1)
-
-        return url
+    def build_property_field(self, field_name, model_class):
+        """
+        Overrides the built-in property field builder to use docstrings as the Swagger
+        help text for fields.
+
+        :param field_name: the name of the property for which the field is being built
+        :param model_class: the ``class`` instance for the Django model
+        :return: the Field subclass to use and the keyword arguments to pass to it
+        """
+
+        klass, kwargs = super().build_property_field(field_name, model_class)
+        kwargs |= {
+            "allow_null": True,
+        }
```

Comment:
Why is this step necessary?

Reply:
This is because `Hit` objects don't have the same properties as the Django media models. In those cases, the serializer can exception and return a 500. Setting `allow_null` to `True` allows the same serializer to handle both search results from ES hits and details from the media model instances.

Reply:
Got it! Would it be possible to add a comment about that here? Or is there somewhere else that behavior is documented? (Just so I know what to look at later when I forget 😅)

Reply:
Added in 0bb2301.